tests/test_pyopener.py

```diff
@@ -48,10 +48,19 @@ def test_opener_fsspec_open(urlpath):
         assert profile["count"] == 3
 
 
+def test_opener_fsspec_fs_open():
+    """Use fsspec filesystem open() as opener."""
+    fs = fsspec.filesystem("file")
+    with rasterio.open("tests/data/RGB.byte.tif", opener=fs.open) as src:
+        profile = src.profile
+        assert profile["driver"] == "GTiff"
+        assert profile["count"] == 3
+
+
 def test_opener_fsspec_fs():
     """Use fsspec filesystem as opener."""
     fs = fsspec.filesystem("file")
-    with rasterio.open("tests/data/RGB.byte.tif", opener=fs.open) as src:
+    with rasterio.open("tests/data/RGB.byte.tif", opener=fs) as src:
```

Comment:
I think this will become the canonical usage: pass in a filesystem object to open the given URI.